pkg/translator/prometheusremotewrite/helper.go

```diff
@@ -105,6 +106,10 @@ func createAttributes(resource pcommon.Resource, attributes pcommon.Map, externa
 	ignoreAttrs []string, logOnOverwrite bool, extras ...string) []prompb.Label {
 	resourceAttrs := resource.Attributes()
 	serviceName, haveServiceName := resourceAttrs.Get(conventions.AttributeServiceName)
+	// SDKs operate with a default service name prefixed with "unknown_service", we should treat such values as undefined.
+	if haveServiceName && strings.HasPrefix(serviceName.AsString(), "unknown_service") {
+		haveServiceName = false
+	}
```

Comment:
Regardless where we land on the rest of this PR, I don't believe that this is an appropriate change.  There is nothing magical about the string "unknown_service" and it could be a valid choice for a service name.  Nothing in the specification indicates that it would be appropriate to ignore the service name attribute if it looks like it might not have been set in the producing SDK.  In fact, the [spec](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/compatibility/prometheus_and_openmetrics.md#resource-attributes-1) _requires_ that collector exporters use the service name to form the job label:

> In the collector Prometheus exporters, the `service.name` and `service.namespace` attributes MUST be combined as `<service.namespace>/<service.name>`, or `<service.name>` if namespace is empty, to form the `job` metric label. 

I understand why it might be attractive to include this change with the other, enabling dropping target info that would only have questionably-identifying service name attributes, but it is what the spec requires.  I also think the spec as it is is correct.  Even with [default SDK resource detection](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/resource/README.md#service) providing the service name as `unknown_service:<process.executable.name>` that is often as uniquely-identifying as an explicitly-provided service name.

Reply:
I've removed this change.